### DISCUSSION TITLE
How to specify an input *directory* for `custom_target` or `generator`?

### DISCUSSION THREAD

**Discussion by ShamrockLee** [mesonbuild/meson#11230] [Q&A] [answered]
When trying to specify a directory `sample_hello.source` as `input` for `custom_target` or `generator.process`, Meson complains

```
../test/samples/meson.build:8:0: ERROR: File sample_hello.source does not exist.
```

How can I feed a directory to my script and produce the output file with Meson?

**Comment by bruchar1** [comment, marked as answer]
A `custom_target` needs files as its input, because it needs to know which modified files will trigger the build.

For a `generator`, you can provide the arguments you want to the script, because it is triggered by its outputs. Therefore, you could provide a directory as an argument to your script.
